sw/cheri/checks/rs485_check.cc

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define CHERIOT_NO_AMBIENT_MALLOC
+#define CHERIOT_NO_NEW_DELETE
+#define CHERIOT_PLATFORM_CUSTOM_UART
+
+#include "../../common/defs.h"
+#include "../common/uart-utils.hh"
+#include "../common/platform-pinmux.hh"
+#include "../common/timer-utils.hh"
+#include <cheri.hh>
+
+#define RS485_UART 2
+
+using namespace CHERI;
+
+[[noreturn]] extern "C" void entry_point(void *rwRoot) {
+  Capability<void> root{rwRoot};
+
+  Capability<volatile uint8_t> pinmux = root.cast<volatile uint8_t>();
+  pinmux.address()                    = PINMUX_ADDRESS;
+  pinmux.bounds()                     = PINMUX_BOUNDS;
+
+  SonataPinmux Pinmux = SonataPinmux(pinmux);
+
+  Pinmux.output_pin_select(SonataPinmux::OutputPin::rs485_tx, 1);
+  Pinmux.block_input_select(SonataPinmux::BlockInput::uart_2_rx, 4);
+
+  Capability<volatile OpenTitanUart> uart = root.cast<volatile OpenTitanUart>();
+  uart.address()                          = UART_ADDRESS;
+  uart.bounds()                           = UART_BOUNDS;
+  uart->init(BAUD_RATE);
+
+  Capability<volatile OpenTitanUart> rs485_uart = root.cast<volatile OpenTitanUart>();
+  rs485_uart.address()                          = UART_ADDRESS + RS485_UART * UART_RANGE;
+  rs485_uart.bounds()                           = UART_BOUNDS;
+  rs485_uart->init(BAUD_RATE);
+
+  write_str(uart, "Writing to RS-485 Uart\r\n");
+
+  uint32_t num = 0;
+  while (1) {
+    write_str(rs485_uart, "Hello from RS-485! Type something and terminate with enter (\\r):\r\n");
+
+    char uart_capture_buf[256];
+    uint32_t chars_seen = 0;
+    char ch;
+
+    do {
+      ch = rs485_uart->blocking_read();
+
+      write_str(uart, "Saw RS-485: 0x");
+      write_hex8b(uart, ch);
+      write_str(uart, "\r\n");
```

Comment:
Agreed, this will fall over quickly if there's transmission happening at any speed, I've just used it for interactive use as a basic receive/transmit test (one that proved useful as I determined I had to let the `tx` line return to idle state before disabling the driver otherwise the transceiver doesn't correctly output idle from the receiver when the RS485 is un-driven).

I was considering writing a higher performance test that used two sonata board along with a PRNG to generate a stream of predictable numbers that can be sent from one board and received by another and checked.